cli/azd/pkg/environment/names.go

```diff
@@ -3,10 +3,20 @@
 
 package environment
 
+import "os"
+
 func GetResourceGroupNameFromEnvVar(env *Environment) string {
+	// First check azd environment
```

Comment:
For some scenarios, like a CD workflow it might be useful to be able to override the azd environment with OS rg.